.github/workflows/test.yml

```diff
@@ -151,47 +151,35 @@ jobs:
   test-emscripten:
     name: Test Emscripten
     runs-on: ubuntu-latest
+    env:
+      PYODIDE_VERSION: '0.21.0-alpha.2'
+      PYTHON_VERSION: '3.10.2'
     steps:
       - uses: actions/checkout@v3
+      - run: |
+            echo "ORIG_PATH=$PATH" >> $GITHUB_ENV
+      - uses: actions/setup-python@v2
+        id: setup-python
+        with:
+          python-version: ${{ env.PYTHON_VERSION }}
       - uses: actions/setup-node@v3
         with:
           node-version: 18
-      - run: |
-          PYODIDE_VERSION=0.21.0-alpha.2
-
-          cd tests
-          npm i pyodide@0.21.0-alpha.2 prettier
-          cd node_modules/pyodide/
-          node ../prettier/bin-prettier.js -w pyodide.asm.js
-          EMSCRIPTEN_VERSION=$(node -p "require('./repodata.json').info.platform.split('_').slice(1).join('.')")
-          PYTHON_VERSION=3.10.2
-
-          echo "PYODIDE_VERSION=$PYODIDE_VERSION" >> $GITHUB_ENV
-          echo "EMSCRIPTEN_VERSION=$EMSCRIPTEN_VERSION" >> $GITHUB_ENV
-          echo "PYTHON_VERSION=$PYTHON_VERSION" >> $GITHUB_ENV
-          echo "ORIG_PATH=$PATH" >> $GITHUB_ENV
       - uses: actions-rs/toolchain@v1
         with:
           profile: minimal
           toolchain: nightly
           components: rust-src
           target: wasm32-unknown-emscripten
           override: true
+      - uses: Swatinem/rust-cache@v1
+      - run: pip install nox
+      - name: Setup Pyodide
+        run: nox -s setup-pyodide
       - uses: mymindstorm/setup-emsdk@v11
         with:
-          version: ${{env.EMSCRIPTEN_VERSION}}
+          version: ${{ env.EMSCRIPTEN_VERSION }}
           actions-cache-folder: emsdk-cache
-      - uses: actions/setup-python@v2
-        id: setup-python
-        with:
-          python-version: ${{env.PYTHON_VERSION}}
-      - run: pip install nox
-      - uses: actions/cache@v3
-        with:
-          path: |
-            tests/pyodide
-          key: ${{ hashFiles('tests/*.js') }} - ${{ hashFiles('noxfile.py') }} - ${{ steps.setup-python.outputs.python-path }}
-      - uses: Swatinem/rust-cache@v1
       - name: Run tests
         run: |
           export PATH=$ORIG_PATH:$PATH
```

Comment:
@hoodmane Could explain why is this needed? Otherwise it throws the following error:

```
warning: no blob constructor, cannot create blobs with mimetypes
warning: no BlobBuilder
/home/runner/work/maturin/maturin/tests/node_modules/pyodide/pyodide.asm.js:5716
          throw ex;
          ^
TypeError: wasm function signature contains illegal type
    at lseek (<anonymous>:wasm-function[11109]:0x55491b)
    at <anonymous>:wasm-function[6841]:0x34ecde
    at method_call_trampoline (/home/runner/work/maturin/maturin/tests/node_modules/pyodide/pyodide.asm.js:8785:33)
    at <anonymous>:wasm-function[1078]:0x137dc0
    at PyObject_VectorcallMethod (<anonymous>:wasm-function[954]:0x13081e)
    at <anonymous>:wasm-function[6926]:0x353e00
    at <anonymous>:wasm-function[6[89](https://github.com/PyO3/maturin/runs/7051097339?check_suite_focus=true#step:10:90)1]:0x35124e
    at <anonymous>:wasm-function[68[90](https://github.com/PyO3/maturin/runs/7051097339?check_suite_focus=true#step:10:91)]:0x35114d
    at <anonymous>:wasm-function[2222]:0x17fe6f
    at method_call_trampoline (/home/runner/work/maturin/maturin/tests/node_modules/pyodide/pyodide.asm.js:8785:33)
```

Reply:
OK, I think it's `setup-emsdk` added an old version of Node to `PATH`.

Reply:
Sorry I didn't leave a comment about that could have saved you the time working this out.